inc/Engine/Common/Queue/AbstractASQueue.php

```diff
@@ -46,8 +48,13 @@ public function schedule_single( $timestamp, $hook, $args = [] ) {
 	 */
 	public function schedule_recurring( $timestamp, $interval_in_seconds, $hook, $args = [] ) {
 		if ( $this->is_scheduled( $hook, $args ) ) {
+			$all = $this->search( [ 'status' => [ ActionScheduler_Store::STATUS_RUNNING, ActionScheduler_Store::STATUS_PENDING ] ] );
+			if ( count( $all ) > 1 ) {
+				$this->cancel_all( $hook, $args );
+			}
```

Comment:
Here I have some comments as follows:-
1. `search` method's second parameter is `$return_format` and its default value is `OBJECT` and we don't need those objects so I believe we can use `ids` instead so we just get the actions IDs and that's good from performance perspective.
2. You need to use Yoda condition like that
    ```if ( 1 < count( $all ) ) {```
3. Use descriptive variable names like `$scheduled_actions`
4. You need to apply the concept of `return early` instead of using nested if conditions to be like that:
```suggestion
			$scheduled_actions = $this->search( [ 'status' => [ ActionScheduler_Store::STATUS_RUNNING, ActionScheduler_Store::STATUS_PENDING ] ] );
			if ( 1 === count( $scheduled_actions ) ) {
				return '';
			}
			
			// Here means that it's greater than 1 as this is enclosed into the condition of is_scheduled.
			$this->cancel_all( $hook, $args );
			return '';
```
5. Add a new condition to the `search` method call for the group to search only for our group (Small optional enhancement)

What do u think about those points?